torch/csrc/jit/serialization/export.cpp

```diff
@@ -433,6 +439,12 @@ void EncoderBase::EncodeGraph(
     bool add_node_names,
     bool use_external_data_format,
     const std::string& onnx_file_path) {
+  // If graph proto size exceed maximum protobuf size of 2GB, set
+  // use_external_data_format to true.
+  if (!use_external_data_format && !onnx_file_path.empty() &&
+      GetGraphProtoSize(graph_proto, graph, initializers) > INT_MAX) {
```

Comment:
oh, I just think of one edge case, if `GetGraphProtoSize` is larger than `UINT_MAX` which is overflow, then this result is undefined. Then it may not be ok to compare with `INT_MAX`. Can we compare `sizes` with `INT_MAX` every time when it tries to add up? How many `sizes += tensor_proto->ByteSizeLong();` happens in a large-size model?

Reply:
That is indeed an edge case, how about we use higher precision type of sizes?